syft/pkg/cataloger/common/cpe/sort_by_specificity.go

```diff
@@ -19,11 +20,30 @@ func (c BySpecificity) Less(i, j int) bool {
 	jScore := weightedCountForSpecifiedFields(c[j])
 
 	if iScore == jScore {
+		if countFieldLength(c[i]) == countFieldLength(c[j]) {
+			// lower index count preferred
+			return dashIndex(c[i]) > dashIndex(c[j])
+		}
+
 		return countFieldLength(c[i]) > countFieldLength(c[j])
 	}
 	return iScore > jScore
 }
 
+func dashIndex(cpe wfn.Attributes) int {
+	count := 0
+	dash := "-"
```

Comment:
nit: This could be a `const` since we never need to assign a new value to it.

Reply:
I need to rename the PR. The bug was `sorted spdx output`, but that was not actually the issue. The issue was that the diffs found when running multiple of the same image were located in the `cpes` array.

Reply:
Updated to: stable sorted CPE array (JSON and SPDX)